api/graphql/types/reservation_unit_option/permissions.py

```diff
@@ -29,5 +29,4 @@ def has_update_permission(cls, instance: ReservationUnitOption, user: AnyUser, i
         if not user.has_staff_permissions:
             return False
 
-        sector = instance.application_section.application.application_round.service_sector
-        return can_manage_service_sectors_applications(user, sector)
+        return can_manage_service_sectors_applications(user)
```

Comment:
Similar to `AllocatedTimeSlotPermission`, the proper check for update permission on `ReservationUnitOption `would now be that the user (if not super user) must have the `CAN_HANDLE_APPLICATIONS` permission on the unit of the option, either from unit or general roles.
(Then can_manage_service_sectors_applications can be dropped from the permission helpers. Or perhaps replace with a new helper, `can_allocate_application` which is scoped at unit?])

Although in this case, should the user of the application be permitted as well, if the application period hasn't passed?

Reply:
Update operations on the reservation unit options are limited to rejecting or locking the option, which a regular user shouldn't be doing. Creation is handled differently, as options are handled in bulk together with application sections, and thus follow its permissions.

I modified the check like all the rests.